go-controller/pkg/ovn/controller/apbroute/external_controller.go

```diff
@@ -47,6 +47,42 @@ type ExternalRouteInfo struct {
 	PodExternalRoutes map[string]map[string]string
 }
 
+type ExternalGatewayRouteInfoCache struct {
+	// External gateway caches
+	// Make them public so that they can be used by the annotation logic to lock on namespaces and share the same external route information
+	ExternalGWCache map[ktypes.NamespacedName]*RouteInfo
+	ExGWCacheMutex  *sync.RWMutex
+	routeInfoLocks  *syncmap.SyncMap[string]
+}
+
+func NewExternalGatewayRouteInfoCache() *ExternalGatewayRouteInfoCache {
+	return &ExternalGatewayRouteInfoCache{
+		ExternalGWCache: make(map[ktypes.NamespacedName]*RouteInfo),
+		ExGWCacheMutex:  &sync.RWMutex{},
+		routeInfoLocks:  syncmap.NewSyncMap[string](),
+	}
```

Comment:
I made some changes on top of this PR to utilize syncmap even more
https://github.com/ovn-org/ovn-kubernetes/pull/3838

Reply:
I left two comments there, but overall it's a noticeable improvement over this approach :smile:

Reply:
Agreed to use your solution over what I did. I'll cherry-pick your commits and add them to this PR.